File: Backend/src/main/java/com/luckyseven/backend/domain/expense/dto/ExpenseResponse.java
package com.luckyseven.backend.domain.expense.dto;

import com.luckyseven.backend.domain.expense.enums.ExpenseCategory;
import com.luckyseven.backend.domain.expense.enums.PaymentMethod;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record ExpenseResponse(
    Long id,
    String description,
    BigDecimal amount,
    ExpenseCategory category,
    Long payerId,
    String payerNickname,
    LocalDateTime createdAt,
    LocalDateTime updatedAt,
    PaymentMethod paymentMethod
) {

}

File: Backend/src/main/java/com/luckyseven/backend/domain/expense/dto/ExpenseResponse.kt
package com.luckyseven.backend.domain.expense.dto

import com.luckyseven.backend.domain.expense.enums.ExpenseCategory
import com.luckyseven.backend.domain.expense.enums.PaymentMethod
import java.math.BigDecimal
import java.time.LocalDateTime


data class ExpenseResponse(
    val id: Long?,
    val description: String,
    val amount: BigDecimal,
    val category: ExpenseCategory,
    val payerId: Long?,
    val payerNickname: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val paymentMethod: PaymentMethod
)
